### PR TITLE
Sentinel Typed Events

### PR DESCRIPTION
**Pull request by asamere**

### CODE REVIEW

File: proto/arkeo/arkeo/events.proto [hunk 1]
```diff
@@ -10,10 +10,14 @@ import "arkeo/arkeo/keeper.proto";
 option go_package = "github.com/arkeonetwork/arkeo/x/arkeo/types";
 
 message EventBondProvider {
-  string creator = 1;
-  string provider = 2 [ (gogoproto.casttype) = "github.com/arkeonetwork/arkeo/common.PubKey" ];
-  string service = 3;
-  string bond = 4 [
+  string provider = 1 [ (gogoproto.casttype) = "github.com/arkeonetwork/arkeo/common.PubKey" ];
+  string service = 2;
+  string bondRel = 3 [
```
Comment (0xean) [inline proto/arkeo/arkeo/events.proto:15]: I must have missed when these variables got introduced, but thoughts on a better / clearer name?  Not really related to this PR, but just caught my eye.

File: sentinel/event_stream.go [hunk 2]
```diff
@@ -180,3 +220,19 @@ func (p Proxy) handleNewBlockHeaderEvent(result tmCoreTypes.ResultEvent) {
 func (p Proxy) isMyPubKey(pk common.PubKey) bool {
 	return pk.Equals(p.Config.ProviderPubKey)
 }
+
+func parseTypedEvent(result tmCoreTypes.ResultEvent, eventType string) (typedEvent proto.Message, err error) {
```
[the remark this answers is not in the document]
Comment (asamere) [reply sentinel/event_stream.go]: nope, fixed

[the remark this answers is not in the document]
Comment (asamere) [reply sentinel/event_stream.go]: nope, fixed

File: proto/arkeo/arkeo/events.proto [hunk 3]
```diff
@@ -38,3 +42,60 @@ message EventModProvider {
...
+  string provider = 1 [ (gogoproto.casttype) = "github.com/arkeonetwork/arkeo/common.PubKey" ];
+  uint64 contractId = 2;
+  string service = 3;
+  string client = 4 [ (gogoproto.casttype) = "github.com/arkeonetwork/arkeo/common.PubKey" ];
+  string delegate = 5 [ (gogoproto.casttype) = "github.com/arkeonetwork/arkeo/common.PubKey" ];
+  ContractType type = 6;
+  int64 height = 7;
+  int64 duration = 8;
+  cosmos.base.v1beta1.Coin rate = 9[ (gogoproto.nullable) = false ];
+  int64 openCost = 10;
```
Comment (0xean) [inline proto/arkeo/arkeo/events.proto:56]: `open_cost`

File: proto/arkeo/arkeo/events.proto [hunk 4]
```diff
@@ -38,3 +42,60 @@ message EventModProvider {
...
+  string deposit = 11 [
+    (cosmos_proto.scalar) = "cosmos.Int",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
+    (gogoproto.nullable) = false
+  ];
+  int64 settlementDuration = 12;
```
Comment (0xean) [inline proto/arkeo/arkeo/events.proto:62]: `settlement_duration`

**Review by 0xean — APPROVED** [review]

**Pull request merged**